### ISSUE TITLE
RTSP Stream Requirements

### ISSUE THREAD

**Issue by rlarranaga** [gtxaspec/wz_mini_hacks#513]
Hello,
Do i need an rtsp enabled firmware to enable rtsp server, or does wz_mini_hacks provides the rtsp functionality?

I have to Wyze V3 cameras on 4.36.9.139, and i installed wz_mini_hacks. I can access the web server, but cannot open the rtsp streams in vlc.

Thanks!

**Comment by pjuhl2313** [comment]
Do you have the correct IP address format? 
It should go:
rtsp://username:password@192.168.xx.xx:port/unicast

**Comment by rlarranaga** [comment]
Yeah, i copied it from the interface, currently i have auth disabled, so it
is like that but with no user:pass@

Thanks

**Comment by Bob-AOL** [comment]
Did you enable rtsp?

**Comment by Bob-AOL** [comment]
What same issue? 

And if you are referring to the Wyze Beta RTSP firmware being required, the answer is no, though I personally prefer it and use wz_mini primarily for the USB/POE feature myself.

You are missing the FPS entry above which should be "20", but may not be a necessary entry for the stream to function. I don't actually know.

**Comment by rlarranaga** [comment]
Let me know if you find a solution for this. I will do the same.

Thanks!

**Comment by Bob-AOL** [comment]
Are you sure wz_mini is even loading?

**Comment by rlarranaga** [comment]
It is definetely serving the web interface

**Comment by dudududodododedede** [comment]
RTSP firmware is not a requirement to have RTSP working on wz_mini. I have RTSP working on v3 firmware version 4.36.9.139 (non-RTSP)

The only things I customized in the config file was:
RTSP_HI_RES_ENABLED
RTSP_HI_RES_ENABLE_AUDIO
RTSP_PASSWORD
(left everything else as defaults)

I also added the SSH key of course to authorized_keys

You should check to make sure you're using the right URL. Easiest way to test is by opening the stream in VLC

**Comment by Bob-AOL** [comment]
SSH is not a requirement and can initially be problematic if set up incorrectly.

The MRL must be correct and the login credentials must be the same as for the camera itself if memory serves me correctly.

VLC viewer can be very picky and downright non-functional at times. I would not rely on it for troubleshooting until knowing the login credentials actually work first.

**Comment by rlarranaga** [comment]
What would be the best way to test rtsp without VLC?

Thanks

**Comment by Bob-AOL** [comment]
It _should_ work in VLC. I use Blue Iris. Another way to sort it out is with tinyCam if you have an Android device handy. It has an entry for Wyze/RTSP and will basically tell you what the problem may be if the stream won't start. A bit of a learning curve, but a handy tool and viewer nonetheless. Works with a Firestick as well.

When using the config editor in the web interface it may not correctly save/apply the file changes during the reboot. Double check for the edits afterward to be sure they were actually implemented.

**Comment by dudududodododedede** [comment]
> What would be the best way to test rtsp without VLC? Thanks


You could try telnetting to the camera on the rtsp port

```
~ user$ telnet 192.168.0.225 8554
Trying 192.168.0.225...
Connected to 192.168.0.225.
Escape character is '^]'.
```

That last line means a connection is successful on the RTSP port 8554

Or if you SSH into the camera you can check `top` or `netstat` to ensure the rtsp process is running and that it's listening on the right port

```
[root@WCV3:~]# netstat -tulnap| grep rtsp
tcp        0      0 0.0.0.0:8554            0.0.0.0:*               LISTEN      695/v4l2rtspserver
tcp        0      0 192.168.0.225:8554      192.168.0.243:55471     ESTABLISHED 695/v4l2rtspserver
tcp        0      0 :::8554                 :::*                    LISTEN      695/v4l2rtspserver
udp        0      0 0.0.0.0:6970            0.0.0.0:*                           695/v4l2rtspserver
udp        0      0 0.0.0.0:6971            0.0.0.0:*                           695/v4l2rtspserver
```
most important is line 1 there

**Comment by Bob-AOL** [comment]
Then there's Analiti -- or a simple port scanner. I would also consider starting over with a new SD card if all else fails.

![Analiti Lrg](https://user-images.githubusercontent.com/110629141/236433015-82f7ed5b-c76d-4bbe-81bd-7b64b6b56bb9.png)
